### DISCUSSION TITLE
Saving data from survey-multi-choice to mySQL database

### DISCUSSION THREAD

**Discussion by holgerMitterer** [jspsych/jsPsych#3258] [Q&A]
Hello everyone,

Since I had lots of problems with saving CSV files, I am trying to save data in a mySQL database via php.
That works, however, for responses from survey-multi-choice, the response is an array,
and I only get the response "ARRAY" in my database. 
(response is varchar length 1000 in the mySQL database)

I couldn't find anything, and since the options are so many
 - Change the php script to linearlize the array (The response variable in SQL is so long for that reason....)
 - Change the variable type of the mySQL database
 - Change the data object send to PHP in Javascript

I am not sure which route to take.

Kind regards,
Holger

**Comment by Nega100** [comment]
What version do you use. Maybe you can add JSON.stringify in question_data and question_order : 
```
  var trial_data = {
    "rt": response_time,
    "reuseAns": JSON.stringify(reuseLetter),
    "stimuli": theNewLetter,
    "accuracy": response_button
  };
```
Hope this help

**Comment by holgerMitterer** [comment]
That's an interesting solution. Thanks. I managed to solve it in the php file that does the prepared staement.

`

 $sql = "INSERT INTO $table VALUES(";

  for($i = 0; $i < count($col_names); $i++){
    $name = $col_names[$i];

    $sql .= ":$name";

    if($i != count($col_names)-1){

      $sql .= ", ";

    }

  }
  $sql .= ");";
  $insertstmt = $conn->prepare($sql);
  for($i=0; $i < count($data_array); $i++){
    for($j = 0; $j < count($col_names); $j++){
      $colname = $col_names[$j];
      if(!isset($data_array[$i][$colname])){
        $insertstmt->bindValue(":$colname", null, PDO::PARAM_NULL);
      } else {
        $temp = $data_array[$i][$colname];
        #check whether it is an array
        if (is_array($temp))
        {
            #I know that there will only ever be one key-value pair here, so this is sufficient.
            foreach ($temp as $key => $value) {
                $result = "$key: $value";
              }
            $temp = $result;
        }else{
            echo("\r\nno array");
        }
        $insertstmt->bindValue(":$colname", $temp);
      }
    }
    $insertstmt->execute();
  }

`

This seems to work.
